imap_processing/hi/l1a/housekeeping.py

```diff
@@ -0,0 +1,92 @@
+"""Unpack IMAP-Hi housekeeping data."""
+import collections
+import dataclasses
+
+import xarray as xr
+
+from imap_processing.cdf.global_attrs import ConstantCoordinates
+from imap_processing.cdf.utils import calc_start_time
+from imap_processing.hi.hi_cdf_attrs import (
+    hi_hk_l1a_attrs,
+    hi_hk_l1a_metadata_attrs,
+)
+
+
+def create_dataset(packets):
+    """Create dataset for each metadata field.
+
+    Parameters
+    ----------
+    packets : list
+        packet list
+
+    Returns
+    -------
+    xr.dataset
+        dataset with all metadata field data in xr.DataArray
+    """
+    metadata_arrays = collections.defaultdict(list)
+    description_arrays = collections.defaultdict(list)
+
+    for data_packet in packets:
+        # Add metadata to array
+        for key, value in data_packet.header.items():
+            # convert key to lower case to match SPDF requirement
+            data_key = key.lower()
+            metadata_arrays.setdefault(data_key, []).append(value.raw_value)
+            # add it once since description should be same for all packets
+            if data_key not in description_arrays:
+                description_arrays[data_key] = (
+                    value.long_description
+                    if value.long_description
+                    else value.short_description
+                )
```

Comment:
Might as well just overwrite it then for simplicity?
```suggestion
            description_arrays[data_key] = value.long_description or value.short_description
```
You can also use the `value or other_value` to avoid the if/else if you want to simplify even further.

Reply:
cool. I wasn't sure which way it was prefer. Will this or pick short description if long description was None?

Reply:
> Will this or pick short description if long description was None?

Yes, it will check `value.long_description` and then fall back to `value.short_description` if `long_description` is None.

```bash
>>> None or "abc"
'abc'
```

Reply:
Just tested and it does get one that's not None.